### PR TITLE
mgr/dashboard: iSCSI management UI

### PR DESCRIPTION
**Pull request by tspmelo**
This PR adds the UI for List, Details, Add and Delete of iSCSI targets.

- [x] References tracker ticket
- [x] Includes tests for new functionality or reproducer for bug
- [x] Check if IQN pattern is correct
- [x] Update to latest version of #25638

### CODE REVIEW

**Comment by tspmelo** [comment]
Added iSCSI status message:
image

**Comment by tspmelo** [comment]
Some screenshots:
image
image
image
image
image
image
image

image
image

**Comment by tspmelo** [comment]
retest this please

**Comment by ricardoasmarques** [comment]
Can we replace/remove `redhat` from the auto-generated target iqn?

screenshot from 2019-01-25 10-49-33

**Comment by tspmelo** [comment]
> Remove `red-hat` from the auto-generated target iqn

Done.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html [hunk 1]
```diff
@@ -0,0 +1,549 @@
...
+              </span>
+            </div>
+
+            <span class="help-block"
+                  *ngIf="targetForm.showError('target_iqn', formDir, 'required')"
+                  i18n>This field is required.</span>
+
+            <span class="help-block"
+                  *ngIf="targetForm.showError('target_iqn', formDir, 'pattern')"
+                  i18n>IQN has wrong patter.</span>
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html]: s/patter./pattern./

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+
+              </div>
+              <span class="help-block"
+                    *ngIf="hasAdvancedSettings(imagesSettings[image])"
+                    i18n>This image has modified settings.</span>
+            </ng-container>
+
+            <span class="help-block"
+                  *ngIf="targetForm.showError('disks', formDir, 'required')"
+                  i18n>At least 1 Image is required.</span>
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html]: s/Image/image/

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html [hunk 3]
```diff
@@ -0,0 +1,549 @@
...
+                          *ngIf="initiator.showError('client_iqn', formDir, 'notUnique')"
+                          i18n>Initiator IQN needs to be unique.</span>
+
+                    <span class="help-block"
+                          *ngIf="initiator.showError('client_iqn', formDir, 'required')"
+                          i18n>This field is required.</span>
+
+                    <span class="help-block"
+                          *ngIf="initiator.showError('client_iqn', formDir, 'pattern')"
+                          i18n>IQN has wrong patter.</span>
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html]:
```suggestion
                          i18n>IQN has wrong pattern.</span>
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html [hunk 4]
```diff
@@ -0,0 +1,549 @@
...
+                          </button>
+                        </span>
+                      </div>
+                      <span class="help-block"
+                            *ngIf="initiator.showError('password', formDir, 'required')"
+                            i18n>This field is required.</span>
+
+                      <span class="help-block"
+                            *ngIf="initiator.showError('password', formDir, 'pattern')"
+                            i18n>Passwords must have a length of 12-16 and
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html]:
```suggestion
                            i18n>Passwords must have a length of 12 to 16 characters and
```
See https://github.com/ceph/ceph/pull/25995/files#diff-37b56750ebccce5725fc786fa1762a59R235

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html [hunk 5]
```diff
@@ -0,0 +1,549 @@
...
+                          </button>
+                        </span>
+                      </div>
+                      <span class="help-block"
+                            *ngIf="initiator.showError('mutual_password', formDir, 'required')"
+                            i18n>This field is required.</span>
+
+                      <span class="help-block"
+                            *ngIf="initiator.showError('mutual_password', formDir, 'pattern')"
+                            i18n>Passwords must have a length of 12-16 and
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html]:
```suggestion
                            i18n>Passwords must have a length of 12 to 16 characters and
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html [hunk 6]
```diff
@@ -0,0 +1,549 @@
...
+
+                    <div class="row">
+                      <div class="col-md-12">
+                        <cd-select [data]="group.getValue('members')"
+                                   [options]="groupMembersSelections"
+                                   [messages]="messages.groupInitiator"
+                                   (selection)="onGroupMemberSelection($event)"
+                                   elemClass="btn btn-default pull-right">
+                          <i class="fa fa-fw fa-plus"></i>
+                          <ng-container i18n>Add Initiator</ng-container>
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html]: Use ``Add Initiator`` or ``Add initiator``, but use the same typing across the whole code. Please adapt other text that uses ``Add|Edit|Create|... xxx``.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.ts [hunk 7]
```diff
@@ -0,0 +1,463 @@
...
+      {
+        noOptions: this.i18n(
+          'There are no initiators available. Please make sure you add an initiator to the target.'
+        )
+      },
+      this.i18n
+    )
+  };
+
+  IQN_REGEX = /^iqn\.(19|20)\d\d-(0[1-9]|1[0-2])\.\D{2,3}(\.[A-Za-z0-9-]+)+(:[A-Za-z0-9-\.]+)*$/;
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.ts]: [A-Za-z] will limit the characters to ASCII only. What about Unicode? Is this limited to ASCII by design? If Unicode is not supported, please add a comment.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-list/iscsi-target-list.component.html [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+               title="iSCSI Targets not available"
+               i18n-title>
+  <ng-container i18n>Please consult the <a href="{{docsUrl}}"
+       target="_blank">documentation</a>
+    on how to configure and enable the iSCSI Targets management functionality.</ng-container>
+
+  <ng-container *ngIf="status">
+    <br>
+    <span i18n>Available information:</span>
+    <pre>{{ status }}</pre>
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-list/iscsi-target-list.component.html]: The pre-formated text does not look nice. I don't have an alternative idea yet, but i think this needs to be polished a little bit.
auswahl_002

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html (hunk 6 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html]: Done, all are lowercase now.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.ts (hunk 7 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.ts]: I copied this regex from openATTIC, so I expect it to be correct.

**Comment by s0nea** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins test dashboard

**Pull request merged**